### PR TITLE
Basic Nextflow module able to run on DNAnexus 

### PR DESCRIPTION
**Pull request by Emma-MacK**
The code is now able to be run from the command line to launch a nextflow job on DNAnexus.

The tool needs a path to the folder of bam, will create file path pairs of corresponding BAMs and BAM.BAIs then use them as an input to the VerifyBamID tool. Bar the path to the bam folder there are parameters held in the config files, including the standard VCF file used, the file types expected for the pairs, and the outputs folder, which will not need to be changed for every run. 

The tool is being tested at https://platform.dnanexus.com/panx/projects/GZgkB004F1B5G56VK7K1bvYZ/data/

- - -
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1)

### CODE REVIEW

**Review by chrispyatt — CHANGES_REQUESTED** [review]
> Reviewed 10 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1)* status: all files reviewed, 5 unresolved discussions (waiting on @Emma-MacK)

___
*[`main.nf` line 7 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlbprBlE8deXdwVnbSZ:-NlbprBlE8deXdwVnbS_:b6f391m) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/main.nf#L7)):*
> ```Nextflow
> 
> // run workflow
> // TO DO: make file pair channel for bam, bai
> ```

looks like this can be removed as it's implemented below?
___
*[`main.nf` line 18 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlbvJ0p5nYw2KqWeDFI:-NlbvJ0p5nYw2KqWeDFJ:bcbhzwd) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/main.nf#L18)):*
> ```Nextflow
> // current run command example
> // dx run applet-Gbjp6p84F1B1VkP35y8Bg20B \
> //-inextflow_pipeline_params="--file_path=dx://project-GZgkB004F1B5G56VK7K1bvYZ:/input_data/1701389_S77_L007_markdup"
> ```

is this an example for running on one sample?

Might be good to have an example for a directory containing multiple input files (though best in a readme)
___
*[`bin/nextflow_code.sh` line 22 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlcD8_J8NE5TsLq4HRB:-NlcD8_J8NE5TsLq4HRC:b-rnrqfm) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/bin/nextflow_code.sh#L22)):*
> ```Shell
> 
> dx download "$vcf_file" -o "$vcf_file_name"
> # dx download "$tool_path" -o "verifyBamID"
> ```

remove if not needed
___
*[`bin/nextflow_code.sh` line 28 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlcD5ujChXyGWYLKNjM:-NlcD5ukBQPRgwaAfW-k:b6xdcvj) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/bin/nextflow_code.sh#L28)):*
> ```Shell
> 
> # get verifybamID
> # chmod -R ${pathToBin}/*
> ```

can this be removed?
___
*[`modules/HELLOWORLD/main.nf` line 6 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlbvgJS44dGVD5DEOMJ:-NlbvgJS44dGVD5DEOMK:bfcm7qw) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/modules/HELLOWORLD/main.nf#L6)):*
> ```Nextflow
>     echo "hello world"
>     """
> }
> ```

not needed?

**Review by Emma-MacK — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1)* status: 7 of 11 files reviewed, 5 unresolved discussions (waiting on @chrispyatt)

___
*[`main.nf` line 7 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlbprBlE8deXdwVnbSZ:-Nm5m-rAB-9UUl6FjB6r:b-896fix) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/main.nf#L7)):*
<details><summary><i>Previously, chrispyatt (Chris Pyatt) wrote…</i></summary><blockquote>

looks like this can be removed as it's implemented below?
</blockquote></details>

Done.
___
*[`main.nf` line 18 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlbvJ0p5nYw2KqWeDFI:-Nm5m135AEVffpoxzBrR:b-896fix) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/main.nf#L18)):*
<details><summary><i>Previously, chrispyatt (Chris Pyatt) wrote…</i></summary><blockquote>

is this an example for running on one sample?

Might be good to have an example for a directory containing multiple input files (though best in a readme)
</blockquote></details>

Done.
___
*[`bin/nextflow_code.sh` line 22 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlcD8_J8NE5TsLq4HRB:-Nm5m41p16j6My-rrTox:b-896fix) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/bin/nextflow_code.sh#L22)):*
<details><summary><i>Previously, chrispyatt (Chris Pyatt) wrote…</i></summary><blockquote>

remove if not needed
</blockquote></details>

Done.
___
*[`bin/nextflow_code.sh` line 28 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlcD5ujChXyGWYLKNjM:-Nm5m4fZ964yftmZLEgd:b-896fix) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/bin/nextflow_code.sh#L28)):*
<details><summary><i>Previously, chrispyatt (Chris Pyatt) wrote…</i></summary><blockquote>

can this be removed?
</blockquote></details>

Done.
___
*[`modules/HELLOWORLD/main.nf` line 6 at r1](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NlbvgJS44dGVD5DEOMJ:-Nm5m6WK5NZzGiSASs4_:b-896fix) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/6af54f2b32ae3ac87cd431304eca608278f51c6e/modules/HELLOWORLD/main.nf#L6)):*
<details><summary><i>Previously, chrispyatt (Chris Pyatt) wrote…</i></summary><blockquote>

not needed?
</blockquote></details>

Done.

**Review by chrispyatt — CHANGES_REQUESTED** [review]
> Reviewed 3 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1)* status: 10 of 11 files reviewed, 1 unresolved discussion (waiting on @Emma-MacK)

___
*[`README.md` line 10 at r2](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NqSwvGA4w2Rptk1nKLR:-NqSwvGA4w2Rptk1nKLS:bxjpbpv) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/77a753e5743baa6b3e8a9c42e3b43fc35c3d58f5/README.md#L10)):*
> ```Markdown
>  git clone <repo>
>  dx select <DNAnexus project>
>  dx build --nextflow (inside the cloned folder)
> ```

I don't think the user needs to know how to build the app. Just what it does and how to run it. The dias_batch one is probably the best example, though this doesn't need to be that long of course.

**Review by Emma-MacK — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1)* status: 9 of 11 files reviewed, 1 unresolved discussion (waiting on @chrispyatt)

___
*[`README.md` line 10 at r2](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1#-NqSwvGA4w2Rptk1nKLR:-Nv76huYA1q-LgLJWBII:be2runu) ([raw file](https://github.com/eastgenomics/VerifyBamID_nextflow/blob/77a753e5743baa6b3e8a9c42e3b43fc35c3d58f5/README.md#L10)):*
<details><summary><i>Previously, chrispyatt (Chris Pyatt) wrote…</i></summary><blockquote>

I don't think the user needs to know how to build the app. Just what it does and how to run it. The dias_batch one is probably the best example, though this doesn't need to be that long of course.
</blockquote></details>

I have removed the section on how to build an app

File: nextflow.config [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
 
+params.vcf_file = "dx://project-Fkb6Gkj433GVVvj73J7x8KbV:file-G08V2gj41zgGYp1K79YZXqx8"
+params.file_type = "*{.bam,.bam.bai}"
+params.file_path = ""
+params.outdir = "./verifybamid_output" //name of output folder
+params.bam_files = "$params.file_path"+"$params.file_type"
+
 //machine type
 process {
     withName:VERIFYBAMID {
```
Comment (Emma-MacK) [inline nextflow.config:12]: maybe test to see if works with lower instance so will be cheaper

**Review by chrispyatt — APPROVED** [review]
> Reviewed 1 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/VerifyBamID_nextflow/1)* status: all files reviewed, 1 unresolved discussion

**Pull request merged**